packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -799,144 +590,75 @@ function get_first_child(block) {
 }
 
 /**
- * @param {Array<import('../../types.js').EachItemBlock>} active_transitions
- * @returns {void}
- */
-function destroy_active_transition_blocks(active_transitions) {
-	var length = active_transitions.length;
-
-	if (length > 0) {
-		var i = 0;
-		var block;
-		var transition;
-
-		for (; i < length; i++) {
-			block = active_transitions[i];
-			transition = block.r;
-			if (transition !== null) {
-				block.r = null;
-				destroy_each_item_block(block, null, false);
-			}
-		}
-
-		active_transitions.length = 0;
-	}
-}
-
-/**
- * @param {import('../../types.js').Block} block
- * @returns {Text | Element | Comment}
- */
-export function get_first_element(block) {
-	const current = block.d;
-
-	if (is_array(current)) {
-		for (let i = 0; i < current.length; i++) {
-			const node = current[i];
-			if (node.nodeType !== 8) {
-				return node;
-			}
-		}
-	}
-
-	return /** @type {Text | Element | Comment} */ (current);
-}
-
-/**
- * @param {import('../../types.js').EachItemBlock} block
+ * @param {import('#client').EachItemBlock} block
  * @param {any} item
  * @param {number} index
  * @param {number} type
  * @returns {void}
  */
-function update_each_item_block(block, item, index, type) {
-	const block_v = block.v;
+function update_block(block, item, index, type) {
 	if ((type & EACH_ITEM_REACTIVE) !== 0) {
-		set(block_v, item);
-	}
-	const transitions = block.s;
-	const index_is_reactive = (type & EACH_INDEX_REACTIVE) !== 0;
-	// Handle each item animations
-	const each_animation = block.a;
-	if (transitions !== null && (type & EACH_KEYED) !== 0 && each_animation !== null) {
-		each_animation(block, transitions);
+		set(block.v, item);
 	}
-	if (index_is_reactive) {
-		set(/** @type {import('../../types.js').Value<number>} */ (block.i), index);
+
+	if ((type & EACH_INDEX_REACTIVE) !== 0) {
+		set(/** @type {import('#client').Value<number>} */ (block.i), index);
 	} else {
 		block.i = index;
 	}
-}
 
-/**
- * @param {import('../../types.js').EachItemBlock} block
- * @param {null | Array<import('../../types.js').Block>} transition_block
- * @param {boolean} apply_transitions
- * @param {any} controlled
- * @returns {void}
- */
-export function destroy_each_item_block(
-	block,
-	transition_block,
-	apply_transitions,
-	controlled = false
-) {
-	const transitions = block.s;
-
-	if (apply_transitions && transitions !== null) {
-		// We might have pending key transitions, if so remove them first
-		for (let other of transitions) {
-			if (other.r === 'key') {
-				transitions.delete(other);
-			}
-		}
-		if (transitions.size === 0) {
-			block.s = null;
-		} else {
-			trigger_transitions(transitions, 'out');
-			if (transition_block !== null) {
-				transition_block.push(block);
-			}
-			return;
-		}
-	}
-	const dom = block.d;
-	if (!controlled && dom !== null) {
-		remove(dom);
-	}
-	destroy_effect(/** @type {import('#client').Effect} */ (block.e));
+	// TODO animations
```

Comment:
what exactly is missing here? the whole "animate keyed each blocks" logic?

Reply:
leftover from before